pkg/agent/controller/traceflow/traceflow_controller_test.go

```diff
@@ -568,8 +848,89 @@ func TestProcessTraceflowItem(t *testing.T) {
 	tfc.mockOFClient.EXPECT().InstallTraceflowFlows(tc.tf.Status.DataplaneTag, tc.tf.Spec.LiveTraffic, tc.tf.Spec.DroppedOnly, tc.receiverOnly, nil, tc.ofPort, crdv1alpha1.DefaultTraceflowTimeout)
 	tfc.mockOFClient.EXPECT().SendTraceflowPacket(tc.tf.Status.DataplaneTag, tc.packet, tc.ofPort, int32(-1))
 
-	tfc.enqueueTraceflow(tc.tf)
-
 	got := tfc.processTraceflowItem()
 	assert.Equal(t, tc.expected, got)
 }
+
+func TestValidateTraceflow(t *testing.T) {
+	tcs := []struct {
+		name               string
+		tf                 *crdv1alpha1.Traceflow
+		antreaProxyEnabled bool
+		expectedErr        string
+	}{
+		{
+			name: "AntreaProxy feature disabled with destination as service",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						Service: "svcTest",
+					},
+				},
+			},
+			expectedErr: "using Service destination requires AntreaProxy feature enabled",
+		},
+		{
+			name: "invalid destination IPv4",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						IP: "192.168.1.300",
+					},
+				},
+			},
+			antreaProxyEnabled: true,
+			expectedErr:        "destination IP is not valid: 192.168.1.300",
+		},
+		{
+			name: "AntreaProxy feature disabled with ClusterIP destination",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						IP: "10.96.1.1",
+					},
+				},
+			},
+			expectedErr: "using ClusterIP destination requires AntreaProxy feature enabled",
+		},
+	}
+
+	for _, tt := range tcs {
+		t.Run(tt.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, features.DefaultFeatureGate, features.AntreaProxy, tt.antreaProxyEnabled)()
+
+			tfc := newFakeTraceflowController(t, nil, []runtime.Object{tt.tf}, nil, nil, nil, nil)
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			tfc.crdInformerFactory.Start(stopCh)
+			tfc.crdInformerFactory.WaitForCacheSync(stopCh)
```

Comment:
I don't believe that you need these lines in order to test the `validateTraceflow` method.